scripts/fill_dcp1_file_metadata/fill_dcp1_metadata.py

```diff
@@ -0,0 +1,188 @@
+
+    # Document ID (file metadata UUID) is outside of ["hits"]["files"], so some data massage
+    def change_datafile_uuid(azul_hit: dict):
+        azul_hit['files'][0]['uuid'] = azul_hit['entryId']
+        return azul_hit
+    files = list(map(change_datafile_uuid, r['hits']))
+
+    # A bit more data massage to return only the file metadata (Easier parsing for later)
+    files = [hit['files'][0] for hit in files]
+
+    return files
+
+
+def parse_azul_metadata(azul_file_list: list) -> dict:
+    """
+    Parse the azul metadata about the files and return a map {uuid: metadata}, excluding DCP1/2 analysis files
+    :param azul_file_list:          File list from azul, output from get_files_metadata_from_azul()
+    :return file_metadata_by_uuid:  Dictionary in the format {file_uuid:file_metadata} for all files in azul_file_list
+    """
+    uuid_metadata_map = {file['uuid']: file for file in azul_file_list}
+
+    filesource_invalid_values = ["HCA Release", "DCP/2 Analysis", "DCP/1 Matrix Service"]
+    uuid_metadata_map_filtered = {uuid: metadata for uuid, metadata in uuid_metadata_map.items()
+                                  if not any([source == metadata['fileSource'] for source in filesource_invalid_values])}
+
+    return uuid_metadata_map_filtered
+
+
+def set_up_ingestapi() -> IngestApi:
+    """
+    Set up ingestApi object, with credentials to access and modify information in the database. URL is not asked because
+    this script can only be run in production (Azul service not available in dev/staging)
+    :return ingest_api: IngestApi object set up with token
+    """
+    ingest_api = IngestApi("https://api.ingest.archive.data.humancellatlas.org")
+    ingest_api.set_token(f"Bearer {INGEST_TOKEN}")
+    return ingest_api
+
+
+def patch_file_metadata(uuid_metadata_map: dict, ingest_api: IngestApi, dry_run: bool):
+    """
+    Send a PUT request to each of the files in the uuid_size_map with the updated size and fileContentType extracted
+    from Azul
+    :param uuid_metadata_map:   {file_uuid: metadata} map of all the files that need to be updated
+    :param ingest_api:          IngestApi object with headers set up
+    :param dry_run:             bool value, if True no PUT operations will be performed (But information will be logged)
+    :return:
+    """
+    for uuid, metadata in uuid_metadata_map.items():
+        # Check file exists in Ingest (Non-organically described matrices are virtually indistinguishable via metadata)
+        try:
+            logging.info(f"Searching for file metadata for file {metadata['name']}")
+            entity = ingest_api.get_entity_by_uuid('files', uuid)
+        except HTTPError:
+            logging.warning(f"Could not find file {metadata['name']} in ingest, please make sure it's not a "
+                            f"non-organically described matrix.More details:")
+            logging.warning(json.dumps(metadata, indent=4, separators=(", ", ": ")))
+            continue
```

Comment:
Why is this a warning? Does the user need to attend to these?
If so, I would collect the uuids in a file that the user can easily inspect & process after this script finishes.

Reply:
Yes, the user needs to attend to these, but a UUID will not give any useful info (the file is not in ingest)

I can create a file with a list of these files, I just thought creating a log and another file may be a bit confusing

Reply:
include every relevant info. pack as csv for easy consumption